lib/commands/publish/npmPublishSemiAtomic.js

```diff
@@ -1,12 +1,19 @@
 var npmUpdateAsLatest = require("./npmUpdateAsLatest");
 var npmPublishAsTemp  = require("./npmPublishAsTemp");
+var execSync          = require("../../utils/execSync");
 
-module.exports = function npmPublishSemiAtomic(packages, packagesLoc, versions, callback) {
+module.exports = function npmPublishSemiAtomic(packages, packagesLoc, versions, flags, callback) {
   // Start by publishing a prerelease that we'll bump later to be more atomic
   npmPublishAsTemp(packages, packagesLoc, function (err) {
     if (err) return callback(err);
 
+    if (flags.canary) {
+      console.log("Reseting git state");
```

Comment:
Err... This really worries me, I don't want to do this since it's a very destructive operation. Is there a way you could pass the canary flag into `npmPublishAsTemp` and avoid having to do this?